connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -150,10 +156,30 @@ private void loadInitialConsumerGroups()
 
     List<String> findConsumerGroups()
             throws InterruptedException, ExecutionException {
-        return listConsumerGroups().stream()
+        List<String> filterGroups = listConsumerGroups().stream()
                 .map(ConsumerGroupListing::groupId)
-                .filter(this::shouldReplicate)
+                .filter(this::shouldReplicateByGroupFilter)
                 .collect(Collectors.toList());
+
+        List<String> checkpointGroups = new LinkedList<>();
+        List<String> irrelevantGroups = new LinkedList<>();
+
+        for (String group : filterGroups) {
+            Set<String> consumeTopics = listConsumerGroupOffsets(group).keySet().stream()
+                    .map(TopicPartition::topic)
+                    .filter(this::shouldReplicateByTopicFilter)
+                    .collect(Collectors.toSet());
+            // As long as a group consumes a topic configured in "topics", it needs to sync for this group.
+            // Although the topic in "topics.exclude" may be consumed in this group, this will be filtered out in the MirrorCheckpointTask.
```

Comment:
This part refers to properties of the [DefaultTopicFilter](https://github.com/apache/kafka/blob/31440b00f3ed8de65f368d41d6cf2efb07ca4a5c/connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultTopicFilter.java), but users can specify a different `TopicFilter` if they'd like. We should probably refer to what the `TopicFilter` does instead of properties specific to one `TopicFilter` implementation:
```suggestion
            // Only perform checkpoints for groups that have offsets for at least one topic
            // that's accepted by the topic filter
```

Reply:
nice suggestion!